libvast/vast/table_slice_column.hpp

```diff
@@ -14,27 +14,64 @@
 #pragma once
 
 #include "vast/fwd.hpp"
-#include "vast/table_slice.hpp"
+#include "vast/view.hpp"
+
+#include <caf/meta/type_name.hpp>
 
 #include <cstdint>
+#include <string_view>
 
 namespace vast {
 
-struct table_slice_column {
-  table_slice_column() {
-  }
+/// Convenience helper for traversing a column of a table slice.
+/// @relates table_slice
+class table_slice_column {
+public:
+  /// Defaulted constructors, destructors, and assignment operators.
+  table_slice_column() noexcept;
+  ~table_slice_column() noexcept;
+  table_slice_column(const table_slice_column&) noexcept;
+  table_slice_column& operator=(const table_slice_column&) noexcept;
+  table_slice_column(table_slice_column&&) noexcept;
+  table_slice_column& operator=(table_slice_column&&) noexcept;
 
-  table_slice_column(table_slice_ptr slice, size_t col)
-    : slice{std::move(slice)}, column{col} {
-    // nop
-  }
-  table_slice_ptr slice;
-  size_t column;
+  /// Construct a view on a column of a table slice.
+  /// @param slice The slice to view.
+  /// @param column The viewed column's index.
+  /// @pre `column < slice->columns()`
+  table_slice_column(table_slice_ptr slice, size_t column) noexcept;
+
+  /// Construct a view on a column of a table slice.
+  /// @param slice The slice to view.
+  /// @param column The viewed column's name.
+  /// @note This constructor fails if the column name is not included in the
+  /// table slices layout. Check for success with `slice() != nullptr`.
+  table_slice_column(table_slice_ptr slice, std::string_view column) noexcept;
+
+  /// @returns the data at given row.
+  /// @pre `row < size()`
+  data_view operator[](size_t row) const;
 
+  /// @returns the number of rows in the column.
+  size_t size() const noexcept;
```

Comment:
To me `rows()` is semantically clearer.

Reply:
Using `size()` and `index()` allows for using `table_slice_row` and `table_slice_column` interchangeably.